### DISCUSSION TITLE
Schema.prototype.obj from extended class

### DISCUSSION THREAD

**Discussion by Kuzzy** [typegoose/typegoose#727] [Q&A] [answered]
[Schema.prototype.obj](https://mongoosejs.com/docs/api/schema.html#schema_Schema-obj) returns props only for base class:

```ts
class BaseSchema {
  @prop()
  test: string
}

class TestSchema extends BaseSchema {
  @prop({
      required: true,
      default: false,
  })
  public someProp!: boolean

  @prop({
      required: true,
  })
  public anotherSomeProp!: string
}

const Test = getModelForClass(TestSchema)

console.log(Test.schema.obj)
 // => { test: { type: [Function: String] } } 

```

I'm not sure if it's by design, but shouldn't it return mixed properties for both classes?

**Comment by hasezoey** [comment, marked as answer]
It should inherit all properties, what are you versions? (typegoose, mongoose, typescript) also maybe try `schema.paths` or `schema.tree`

---

Edit: i have now looked at the documentation for `obj`, because i never actually understood what that was for (considering `tree` and `paths` exists), but now that i know that it `The original object passed to the schema constructor`, it explains why your properties do not appear there, it is because typegoose does not create schemas with the first parameter to the schema, but via the `.add` function (which does not appear in the `obj` as seen in the documentation), but it seems mongoose when cloning a schema (which is being done when a extending classes) acts like having everything passed in the first parameter to `new Schema`

**Reply by Kuzzy** [reply]
OK, thanks for explanation. I think I can use `paths` instead `obj` for my needs. But do you plan implement behaviour for `obj` like mongoose do? If not, maybe it would be good to notice about this difference somewhere in documentation?

**Reply by hasezoey** [reply]
> But do you plan implement behaviour for obj like mongoose do? If not, maybe it would be good to notice about this difference somewhere in documentation?

no i do not plan to change this, because `.add` is used, also the `obj` behaves the same as in mongoose, which should not really need anything, though PRs are welcome
